api/clients/v2/coretypes/eigenda_cert.go

```diff
@@ -95,6 +95,18 @@ type EigenDACert interface {
 	Version() CertificateVersion
 }
 
+
+type RetrievableEigenDACert interface {
```

Comment:
Given that there is both `EigenDACert` and `RetrievableEigenDACert`, I think a doc explaining the purpose of each would be helpful. i.e. why do these need to be separate?

TBH I don't understand the benefit of having two interfaces here I read @samlaf 's [comment](https://github.com/Layr-Labs/eigenda/pull/1607#discussion_r2111549815), but I'm not fully following. Why can't we just have an `EigenDACert` interface, which has all of these methods?

I'm approving, since I don't see anything necessary wrong with the current approach, so if you guys have agreement then that works for me too

Reply:
Can we please just get this through and think about a more optimised solution down the line?